specs/dotsigstore_bundle/pb/bundle.proto

```diff
@@ -0,0 +1,167 @@
+syntax = "proto3";
+package sigstore;
+
+// https://raw.githubusercontent.com/secure-systems-lab/dsse/9c813476bd36de70a5738c72e784f123ecea16af/envelope.proto
+import "envelope.proto";
+
+option go_package = "github.com/sigstore/xxx/bundle";
+option java_package = "com.github.sigstore.xxx.bundle";
+
+// Notes on versioning.
+// The primary message ('Bundle') MUST be versioned, by populating the
+// 'media_type' field. Semver-ish (only major/minor versions) scheme MUST
+// be used. The current version as specified by this file is:
+// application/vnd.dev.sigstore.bundle.v0.1+json
+// The semantic version is thus 'v0.1'.
+
+// Only a subset of the secure hash standard algorithms are supported.
+// See https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf for more
+// details.
+enum HashAlgorithm {
+        SHA2_256 = 0;
+        SHA2_512 = 1;
+}
+
+// HashOutput captures a digest of a 'message' and the corresponding hash
+// algorithm used.
+message HashOutput {
+        HashAlgorithm algorithm = 1;
+        // This is the raw octets of the message digest as computed by
+        // the hash algorithm.
+        bytes digest = 2;
+}
+
+// MessageSignature stores the computed signature over a message's (generic
+// octet sequence) digest.
+message MessageSignature {
+        HashOutput payload_digest = 1;
+        bytes signature = 2;
+}
+
+// RekorKindVersion contains the entry's kind and api version.
+message RekorKindVersion {
+        // Kind is the type of entry being stored in the log.
+        // See here for a list: https://github.com/sigstore/rekor/tree/main/pkg/types
+        string kind = 3;
+        // The specific api version of the type.
+        string version = 4;
+}
+
+// InclusionProof is the proof returned from the transparency log. Can
+// be used for on line verification against the log.
+message InclusionProof {
+        // The index of the entry in the log.
+        int64 log_index = 1;
+        // The hash value stored at the root of the merkle tree at the time
+        // the proof was generated, hex encoded.
+        string root_hash = 2;
+        // The size of the merkle tree at the time the proof was generated.
+        int64 tree_size = 3;
+        // A list of hashes required to compute the inclusion proof, sorted
+        // in order from leaf to root. Each hash value is hex encoded.
+        repeated string hashes = 4;
```

Comment:
Agree, the data types now are a raw copy of the response from Rekor. I think we should first get agreement on the data and the structure, then we can take a stab at the data type/encoding. I have deliberately now used the format as returned from Rekor.

Reply:
+1, we don't need to decide encoding questions now.

*But* I think it's worth putting in placeholders to make sure we actually address it later. If undocumented, this could cause big headaches, and I'm personally really likely to forget to come back to this 😄

Reply:
I added a task list in the PR description to updated this according to the outcome of https://github.com/sigstore/community/discussions/136.
I think we can resolve this now?

Reply:
> sigstore/sigstore and be shared by Fulcio/Rekor/clients for maximum consistency.

Totally agree, I listed this as a benefit in this discussion too: https://github.com/sigstore/community/discussions/136

I'll update the task list to actually capture this.